File: Data Structures & Algorithms/duplicate-integer/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {

        const seen = new Set();

        for(const num of nums) {
            if(seen.has(num)) {
                return true;
            } else {
                seen.add(num);
            }
        }

        return false;

    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-4.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        
        count = 0
        prefix_sum = 0
        prefix_sums = { 0: 1 }

        for num in nums:
            prefix_sum += num

            if prefix_sum - k in prefix_sums:
                count += prefix_sums[prefix_sum - k]

            if prefix_sum not in prefix_sums:
                prefix_sums[prefix_sum] = 0
            
            prefix_sums[prefix_sum] += 1
             
        return count